cli/cmd/test_connection.go

```diff
@@ -72,22 +72,42 @@ func updateSyncTestConnectionStatus(ctx context.Context, logger zerolog.Logger,
 	}
 
 	log.Info().Str("status", string(status)).Msg("Sending sync test connection to API")
-	requestBody := cloudquery_api.UpdateSyncTestConnectionJSONRequestBody{
-		Status: status,
-	}
-	if failedTestResult != nil {
-		requestBody.FailureCode = &failedTestResult.FailureCode
-		requestBody.FailureReason = &failedTestResult.FailureDescription
-	}
-	res, err := apiClient.UpdateSyncTestConnectionWithResponse(ctx, teamName, syncTestConnectionUUID, requestBody)
-	if err != nil {
-		log.Warn().Err(err).Msg("Failed to send sync test connection to API")
-		return
+
+	var statusCode int
+	switch os.Getenv("_CQ_SYNC_TEST_CONNECTION_KIND") {
+	case "source":
+		requestBody := cloudquery_api.UpdateSyncTestConnectionForSyncSourceJSONRequestBody{
+			Status: status,
+		}
+		if failedTestResult != nil {
+			requestBody.FailureCode = &failedTestResult.FailureCode
+			requestBody.FailureReason = &failedTestResult.FailureDescription
+		}
+		res, err := apiClient.UpdateSyncTestConnectionForSyncSourceWithResponse(ctx, teamName, syncTestConnectionUUID, requestBody)
+		if err != nil {
+			log.Warn().Err(err).Msg("Failed to send sync test connection result to API")
+			return
+		}
+		statusCode = res.StatusCode()
+	default:
+		requestBody := cloudquery_api.UpdateSyncTestConnectionForSyncDestinationJSONRequestBody{
+			Status: status,
+		}
+		if failedTestResult != nil {
+			requestBody.FailureCode = &failedTestResult.FailureCode
+			requestBody.FailureReason = &failedTestResult.FailureDescription
+		}
+		res, err := apiClient.UpdateSyncTestConnectionForSyncDestinationWithResponse(ctx, teamName, syncTestConnectionUUID, requestBody)
+		if err != nil {
+			log.Warn().Err(err).Msg("Failed to send sync test connection result to API")
+			return
+		}
+		statusCode = res.StatusCode()
 	}
-	if res.StatusCode() != http.StatusOK {
-		log.Warn().Str("status", res.Status()).Int("code", res.StatusCode()).Msg("Failed to send test connection to API")
+	if statusCode != http.StatusOK {
+		log.Warn().Str("status", string(status)).Int("code", statusCode).Msg("Failed to send test connection result to API")
```

Comment:
There's a tiny inconsistency fix here - `res.Status()` is the response message but `string(status)` is the status of the test conn we've tried updating it to.